libcodechecker/server/client_db_access_server.py

```diff
@@ -262,68 +269,92 @@ def do_POST(self):
-            request_endpoint = self.path.replace("/{0}".format(product_name),
-                                                 "", 1)
+            product_endpoint, api_ver, request_endpoint = \
+                routing.split_client_POST_request(self.path)
 
             product = None
-            if product_name:
+            if product_endpoint:
                 # The current request came through a product route, and not
                 # to the main endpoint.
-                product = self.server.get_product(product_name)
+                product = self.server.get_product(product_endpoint)
                 if product and not product.connected:
                     # If the product is not connected, try reconnecting...
                     LOG.debug("Request's product '{0}' is not connected! "
-                              "Attempting reconnect...".format(product_name))
+                              "Attempting reconnect..."
+                              .format(product_endpoint))
                     product.connect()
 
                     if not product.connected:
                         # If the reconnection fails, send an error to the user.
+                        LOG.debug("Product reconnection failed.")
                         self.send_error(  # 500 Internal Server Error
                             500, "Product '{0}' database connection failed!"
-                                 .format(product_name))
+                                 .format(product_endpoint))
                         return
                 elif not product:
+                    LOG.debug("Requested product does not exist.")
                     self.send_error(
                         404, "The product {0} does not exist."
-                             .format(product_name))
+                             .format(product_endpoint))
                     return
 
-            if request_endpoint == '/Authentication':
-                auth_handler = ThriftAuthHandler(self.server.manager,
-                                                 auth_session,
-                                                 self.server.config_session)
-                processor = codeCheckerAuthentication.Processor(auth_handler)
-            elif request_endpoint == '/Products':
-                prod_handler = ThriftProductHandler(
-                    self.server,
-                    auth_session,
-                    self.server.config_session,
-                    product,
-                    version)
-                processor = codeCheckerProductService.Processor(prod_handler)
-            elif request_endpoint == '/CodeCheckerService':
-                # This endpoint is a product's report_server.
-                if not product:
-                    self.send_error(  # 404 Not Found
-                        404, "The specified product '{0}' does not exist!"
-                             .format(product_name))
-                    return
+            if api_ver == '6.0':
```

Comment:
Shouldn't we move out this v6 specific routing to a separate module? If we will have another version it could be better separated.

Reply:
That was my original idea too (hence the `router` module), but it turns out that API handlers for different versions could end up requiring different arguments to their constructors.

Which, with a function in a separate module, would only introduces an unnecessary extra function call. Instead of

```py
  if api_ver == 'ver':
    if endpoint == 'A': 
      # ...
    elif endpoint == 'B':
     #
```

You would get

```
  if api_ver == 'ver':
    handler = make_handler(endpoint, arg1, arg2, arg3, arg4)
```

Where some of `argX` might not exist or is not used, eventually coming back to the same level of bloat.